In [1]:
import json
file = json.load(open('/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/[라벨]gbt_fish_dtset/gbt_fish_dtset4.json'))
file.keys() # categories', 'images', 'annotations
file['categories']

[{'name': 'Olive flounder', 'supercategory': 'fish', 'id': 1},
 {'name': 'Korea rockfish', 'supercategory': 'fish', 'id': 2},
 {'name': 'Red seabream', 'supercategory': 'fish', 'id': 3},
 {'name': 'Black porgy', 'supercategory': 'fish', 'id': 4},
 {'name': 'Rock bream', 'supercategory': 'fish', 'id': 5}]

In [2]:
file['images']

[{'light': 'L03',
  'file_name': './382255_objt_of_2021-01-07_12-11-21-00_001.JPG',
  'width': 2704,
  'date_captured': '2021-01-28 10:47:31',
  'id': 0,
  'height': 1520},
 {'light': 'L03',
  'file_name': './617094_objt_of_2021-02-07_09-46-19-33_003.JPG',
  'width': 2704,
  'date_captured': '2021-02-12 16:50:57',
  'id': 1,
  'height': 1520},
 {'light': 'L02',
  'file_name': './139221_objt_rs_2020-12-17_10-57-19-33_002.JPG',
  'width': 2704,
  'date_captured': '2020-12-22 13:07:08',
  'id': 2,
  'height': 1520},
 {'light': 'L02',
  'file_name': './214655_objt_rs_2021-01-04_12-12-53-33_002.jpg',
  'width': 2704,
  'date_captured': '2021-01-12 14:08:08',
  'id': 3,
  'height': 1520},
 {'light': 'L02',
  'file_name': './318109_objt_rs_2021-01-18_13-46-48-00_002.JPG',
  'width': 2704,
  'date_captured': '2021-01-25 14:01:03',
  'id': 4,
  'height': 1520},
 {'light': 'L03',
  'file_name': './474148_objt_kr_2021-01-27_18-08-07-00_002.JPG',
  'width': 2704,
  'date_captured': '2021-02-04 16:

In [3]:
file['annotations']

[{'diseases_bbox': [],
  'category_id': 1,
  'iscrowd': 1,
  'bbox': [810, 254, 536, 1253],
  'id': 0,
  'attribute': 'objt',
  'diseases_desc': '',
  'image_id': 0,
  'keypoints': [1089, 1495, 1089, 445, 857, 986, 1280, 993],
  'gd': 317,
  'diseases_exist': False},
 {'diseases_bbox': [],
  'category_id': 1,
  'iscrowd': 1,
  'bbox': [1325, 289, 487, 1103],
  'id': 1,
  'attribute': 'objt',
  'diseases_desc': '',
  'image_id': 1,
  'keypoints': [1648, 1393, 1508, 424, 1754, 954, 1404, 1002],
  'gd': 348,
  'diseases_exist': False},
 {'diseases_bbox': [],
  'category_id': 3,
  'iscrowd': 1,
  'bbox': [1088, 499, 1425, 516],
  'id': 2,
  'attribute': 'objt',
  'diseases_desc': '',
  'image_id': 2,
  'keypoints': [1090, 739, 2213, 839, 1500, 510, 1464, 941],
  'gd': 155,
  'diseases_exist': False},
 {'diseases_bbox': [],
  'category_id': 3,
  'iscrowd': 1,
  'bbox': [56, 698, 1127, 453],
  'id': 3,
  'attribute': 'objt',
  'diseases_desc': '',
  'image_id': 3,
  'keypoints': [1183, 900, 

* json 내에서 필요한 정보
이미지 너비 2704 , 높이 1520으로 고정 (96dpi)

https://www.pixelto.net/px-to-cm-converter

json 내 keys => categories , images , annotation

** categories
id : 1(넙치(광어)) Olive flounder
id : 2(우럭(조피볼락)) Korea rockfish
id : 3(참돔) Red seabream
id : 4(감성돔) Black porgy
id : 5(돌돔) Rock bream

** images
- id : annotation의 id와 동일
- file_name : annotation의 id와 일치하는 images의 id의 파일명을 추출

** annotations
- category_id : 어종 판별
- id : images의 id와 일치하는 파일명을 추출해야함
- bbox : 3번째 값(너비) => 가져와서 환산해야함
- diseases_exist': False (질병 없는 애들만)


** 전처리 전에 데이터 수집용
1. id별로 묶고 id별 file_name 찾아서 어종별 폴더 만들고 거기에 이미지 저장
2. diseases_exist': False => 질병 없는 애들만
3. bbox : 0.026458333 * 기존 width(3번째 값) -> 각 어종별 포획금지 체장 이상인 것만


* 최종으로 나와야 되는 라벨링 파일(coco json)
- coco json 파일을 바꿔야 되는데
- json => dictionary : keys => category_id / bbox : 0.026458333 * 기존 width(3번째 값)  

* 어종분류 (1번째 모델)
- category_id => 어종 분류(5종)

* score_model 
- bbox 계산된 값



VGG16 / RESNET50 끝내기
코랩 1만원 이하 => GPU 좋음


*KKM
alexnet
googlenet

*PHS
YOLO
MMDetection

In [4]:
file['images'][0]['file_name']

'./382255_objt_of_2021-01-07_12-11-21-00_001.JPG'

In [5]:
len(file['images'])

100400

In [6]:
import pandas as pd

df = pd.DataFrame()

# file_name = []

# for i in len(range(file['images'])):
#     file_name.append(file['images'][i]['file_name'])
    
# print(file_name)

In [7]:
data1 = pd.json_normalize(file['images'])
data1.head()

light                                        file_name  width  \
0   L03  ./382255_objt_of_2021-01-07_12-11-21-00_001.JPG   2704   
1   L03  ./617094_objt_of_2021-02-07_09-46-19-33_003.JPG   2704   
2   L02  ./139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704   
3   L02  ./214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704   
4   L02  ./318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704   

         date_captured  id  height  
0  2021-01-28 10:47:31   0    1520  
1  2021-02-12 16:50:57   1    1520  
2  2020-12-22 13:07:08   2    1520  
3  2021-01-12 14:08:08   3    1520  
4  2021-01-25 14:01:03   4    1520

In [8]:
data2 = pd.json_normalize(file['annotations'])
data2.head()

diseases_bbox  category_id  iscrowd                    bbox  id attribute  \
0            []            1        1   [810, 254, 536, 1253]   0      objt   
1            []            1        1  [1325, 289, 487, 1103]   1      objt   
2            []            3        1  [1088, 499, 1425, 516]   2      objt   
3            []            3        1    [56, 698, 1127, 453]   3      objt   
4            []            3        1   [368, 329, 1030, 391]   4      objt   

  diseases_desc  image_id                                       keypoints  \
0                       0    [1089, 1495, 1089, 445, 857, 986, 1280, 993]   
1                       1  [1648, 1393, 1508, 424, 1754, 954, 1404, 1002]   
2                       2    [1090, 739, 2213, 839, 1500, 510, 1464, 941]   
3                       3      [1183, 900, 285, 966, 859, 727, 888, 1056]   
4                       4       [370, 555, 1267, 559, 779, 338, 776, 696]   

    gd  diseases_exist  
0  317           False  
1  348           False  
2  155           False  
3  173           False  
4  553           False

In [9]:
data2['fish_size'] = None

print(data2['bbox'][0][2]*0.026458333)
print(data2['bbox'][1][2]*0.026458333)

print(len(data2['bbox']))

14.181666488000001
12.885208171
100400


In [10]:
for i in range(len(data2['bbox'])):
    data2['fish_size'][i] = data2['bbox'][i][2]*0.026458333
    
data2['fish_size'].head()

/var/folders/tn/bfvnwzjj4xn3d7bqc2nx96nr0000gn/T/ipykernel_16754/1959285849.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data2['fish_size'][i] = data2['bbox'][i][2]*0.026458333


0    14.181666
1    12.885208
2    37.703125
3    29.818541
4    27.252083
Name: fish_size, dtype: object

In [11]:
data3 = pd.concat([data1 , data2] , axis=1)
data3.head()

light                                        file_name  width  \
0   L03  ./382255_objt_of_2021-01-07_12-11-21-00_001.JPG   2704   
1   L03  ./617094_objt_of_2021-02-07_09-46-19-33_003.JPG   2704   
2   L02  ./139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704   
3   L02  ./214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704   
4   L02  ./318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704   

         date_captured  id  height diseases_bbox  category_id  iscrowd  \
0  2021-01-28 10:47:31   0    1520            []            1        1   
1  2021-02-12 16:50:57   1    1520            []            1        1   
2  2020-12-22 13:07:08   2    1520            []            3        1   
3  2021-01-12 14:08:08   3    1520            []            3        1   
4  2021-01-25 14:01:03   4    1520            []            3        1   

                     bbox  id attribute diseases_desc  image_id  \
0   [810, 254, 536, 1253]   0      objt                       0   
1  [1325, 289, 487, 1103]   1      objt                       1   
2  [1088, 499, 1425, 516]   2      objt                       2   
3    [56, 698, 1127, 453]   3      objt                       3   
4   [368, 329, 1030, 391]   4      objt                       4   

                                        keypoints   gd  diseases_exist  \
0    [1089, 1495, 1089, 445, 857, 986, 1280, 993]  317           False   
1  [1648, 1393, 1508, 424, 1754, 954, 1404, 1002]  348           False   
2    [1090, 739, 2213, 839, 1500, 510, 1464, 941]  155           False   
3      [1183, 900, 285, 966, 859, 727, 888, 1056]  173           False   
4       [370, 555, 1267, 559, 779, 338, 776, 696]  553           False   

   fish_size  
0  14.181666  
1  12.885208  
2  37.703125  
3  29.818541  
4  27.252083

In [12]:
data3 = data3.drop(columns=['light', 'date_captured' ,'diseases_bbox' , 'attribute', 'iscrowd'])

In [13]:
data3 = data3.drop(columns=['image_id' , 'keypoints' , 'gd' , 'id' , 'id'])

In [14]:
# 질병이 없는 것만 다시 추림

data4 = data3[data3['diseases_exist'] == False]
data4

file_name  width  height  \
0       ./382255_objt_of_2021-01-07_12-11-21-00_001.JPG   2704    1520   
1       ./617094_objt_of_2021-02-07_09-46-19-33_003.JPG   2704    1520   
2       ./139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704    1520   
3       ./214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704    1520   
4       ./318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704    1520   
...                                                 ...    ...     ...   
100395  ./634921_objt_bp_2021-02-08_09-16-37-33_002.JPG   2704    1520   
100396  ./634962_objt_bp_2021-02-08_09-15-57-33_002.JPG   2704    1520   
100397  ./634638_objt_bp_2021-02-08_09-12-35-33_002.JPG   2704    1520   
100398  ./634616_objt_bp_2021-02-08_09-10-00-66_006.JPG   2704    1520   
100399  ./633922_objt_bp_2021-02-08_09-03-55-00_001.JPG   2704    1520   

        category_id                    bbox diseases_desc  diseases_exist  \
0                 1   [810, 254, 536, 1253]                         False   
1                 1  [1325, 289, 487, 1103]                         False   
2                 3  [1088, 499, 1425, 516]                         False   
3                 3    [56, 698, 1127, 453]                         False   
4                 3   [368, 329, 1030, 391]                         False   
...             ...                     ...           ...             ...   
100395            4    [43, 625, 1421, 625]                         False   
100396            4    [968, 284, 661, 247]                         False   
100397            4  [1039, 411, 1183, 448]                         False   
100398            4   [193, 361, 1112, 384]                         False   
100399            4    [617, 363, 825, 349]                         False   

        fish_size  
0       14.181666  
1       12.885208  
2       37.703125  
3       29.818541  
4       27.252083  
...           ...  
100395  37.597291  
100396  17.488958  
100397  31.300208  
100398  29.421666  
100399  21.828125  

[100290 rows x 8 columns]

In [15]:
data4 = data4.drop(columns=['diseases_desc' , 'diseases_exist'])

In [16]:
# id : 1(넙치(광어)) Olive flounder 금지체장 : 35
# id : 2(우럭(조피볼락)) Korea rockfish 23
# id : 3(참돔) Red seabream 금지체장 : 24
# id : 4(감성돔) Black porgy 25
# id : 5(돌돔) Rock bream 24
data4

file_name  width  height  \
0       ./382255_objt_of_2021-01-07_12-11-21-00_001.JPG   2704    1520   
1       ./617094_objt_of_2021-02-07_09-46-19-33_003.JPG   2704    1520   
2       ./139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704    1520   
3       ./214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704    1520   
4       ./318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704    1520   
...                                                 ...    ...     ...   
100395  ./634921_objt_bp_2021-02-08_09-16-37-33_002.JPG   2704    1520   
100396  ./634962_objt_bp_2021-02-08_09-15-57-33_002.JPG   2704    1520   
100397  ./634638_objt_bp_2021-02-08_09-12-35-33_002.JPG   2704    1520   
100398  ./634616_objt_bp_2021-02-08_09-10-00-66_006.JPG   2704    1520   
100399  ./633922_objt_bp_2021-02-08_09-03-55-00_001.JPG   2704    1520   

        category_id                    bbox  fish_size  
0                 1   [810, 254, 536, 1253]  14.181666  
1                 1  [1325, 289, 487, 1103]  12.885208  
2                 3  [1088, 499, 1425, 516]  37.703125  
3                 3    [56, 698, 1127, 453]  29.818541  
4                 3   [368, 329, 1030, 391]  27.252083  
...             ...                     ...        ...  
100395            4    [43, 625, 1421, 625]  37.597291  
100396            4    [968, 284, 661, 247]  17.488958  
100397            4  [1039, 411, 1183, 448]  31.300208  
100398            4   [193, 361, 1112, 384]  29.421666  
100399            4    [617, 363, 825, 349]  21.828125  

[100290 rows x 6 columns]

In [20]:
data4['file_name2'] = None

for i in range(100290):
    data4['file_name2'][i] = data4['file_name'].str[i][2:]


/var/folders/tn/bfvnwzjj4xn3d7bqc2nx96nr0000gn/T/ipykernel_12483/2959163152.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data4['file_name2'][i] = data4['file_name'].str[i][2:]
/var/folders/tn/bfvnwzjj4xn3d7bqc2nx96nr0000gn/T/ipykernel_12483/2959163152.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data4['file_name2'][i] = data4['file_name'].str[i][2:]
/var/folders/tn/bfvnwzjj4xn3d7bqc2nx96nr0000gn/T/ipykernel_12483/2959163152.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/use

KeyboardInterrupt: 

In [17]:
data4.to_csv('dataset4.csv' , index=False)

In [18]:
import pandas as pd
import openpyxl

data4.to_excel('dataset4.xlsx', index=False)

In [20]:
import pandas as pd

df_data4 = pd.read_csv('dataset4 - [csv]dataset4.csv', encoding='cp949')
df_data4.drop("file_name", axis=1, inplace=True)
df_data4.rename(columns={'file_name.1':'file_name'}, inplace=True)
df_data4[:10]

file_name  width  height  category_id  \
0  382255_objt_of_2021-01-07_12-11-21-00_001.JPG   2704    1520            1   
1  617094_objt_of_2021-02-07_09-46-19-33_003.JPG   2704    1520            1   
2  139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704    1520            3   
3  214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704    1520            3   
4  318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704    1520            3   
5  474148_objt_kr_2021-01-27_18-08-07-00_002.JPG   2704    1520            2   
6  603307_objt_of_2021-02-04_12-06-12-33_003.JPG   2704    1520            1   
7  535244_objt_kr_2021-02-01_13-58-13-00_001.JPG   2704    1520            2   
8  390482_objt_rb_2021-01-21_23-33-39-66_004.JPG   2704    1520            5   
9  314049_objt_bp_2021-01-20_10-35-43-66_003.JPG   2704    1520            4   

                     bbox  fish_size  
0   [810, 254, 536, 1253]  14.181666  
1  [1325, 289, 487, 1103]  12.885208  
2  [1088, 499, 1425, 516]  37.703125  
3    [56, 698, 1127, 453]  29.818541  
4   [368, 329, 1030, 391]  27.252083  
5   [687, 787, 1240, 546]  32.808333  
6   [1126, 236, 662, 815]  17.515416  
7    [41, 760, 1016, 292]  26.881666  
8   [453, 703, 1429, 634]  37.808958  
9   [571, 776, 1268, 606]  33.549166

In [21]:
df_data4.fish_size.dtype

dtype('float64')

In [22]:
dtset4_광어 = df_data4[(df_data4['category_id']==1)&(df_data4['fish_size']>=35)]
dtset4_광어

file_name  width  height  \
125     546105_objt_of_2021-02-04_09-08-11-33_002.JPG   2704    1520   
210     571383_objt_of_2021-02-04_10-36-22-00_001.JPG   2704    1520   
353     446970_objt_of_2021-01-26_09-32-39-00_001.JPG   2704    1520   
539     472639_objt_of_2021-01-28_11-20-24-00_001.JPG   2704    1520   
548     458711_objt_of_2020-12-22_13-38-02-33_002.JPG   2704    1520   
...                                               ...    ...     ...   
98638   436161_objt_of_2020-12-22_10-12-08-00_001.JPG   2704    1520   
98746   498778_objt_of_2021-02-03_10-16-04-66_003.JPG   2704    1520   
99155   589882_objt_of_2021-02-06_09-01-05-00_001.JPG   2704    1520   
100053  547135_objt_of_2021-02-04_09-11-29-33_002.JPG   2704    1520   
100159  545972_objt_of_2021-02-04_09-02-05-00_001.JPG   2704    1520   

        category_id                    bbox  fish_size  
125               1   [543, 302, 1745, 741]  46.169791  
210               1   [633, 256, 1323, 602]  35.004375  
353               1   [589, 542, 1371, 794]  36.274375  
539               1   [162, 382, 1437, 609]  38.020625  
548               1   [823, 418, 1460, 768]  38.629166  
...             ...                     ...        ...  
98638             1  [1091, 144, 1446, 643]  38.258750  
98746             1   [758, 832, 1345, 590]  35.586458  
99155             1   [643, 231, 1521, 645]  40.243124  
100053            1  [1078, 194, 1485, 574]  39.290625  
100159            1   [874, 673, 1379, 613]  36.486041  

[687 rows x 6 columns]

In [23]:
dtset4_우럭 = df_data4[(df_data4['category_id']==2)&(df_data4['fish_size']>=23)]
dtset4_우럭

file_name  width  height  \
5      474148_objt_kr_2021-01-27_18-08-07-00_002.JPG   2704    1520   
7      535244_objt_kr_2021-02-01_13-58-13-00_001.JPG   2704    1520   
15     533442_objt_kr_2021-02-01_12-50-52-00_001.JPG   2704    1520   
16     569465_objt_kr_2021-02-05_12-14-33-66_003.JPG   2704    1520   
18     506537_objt_kr_2021-01-29_13-50-43-66_003.JPG   2704    1520   
...                                              ...    ...     ...   
99472  561396_objt_kr_2021-02-05_14-14-32-33_002.JPG   2704    1520   
99493  543141_objt_kr_2021-02-05_09-36-36-33_002.JPG   2704    1520   
99497  404803_objt_kr_2021-01-25_10-27-36-00_001.JPG   2704    1520   
99500  419587_objt_kr_2021-01-25_14-26-33-00_001.JPG   2704    1520   
99506  247156_objt_kr_2021-01-13_10-41-45-33_002.JPG   2704    1520   

       category_id                    bbox  fish_size  
5                2   [687, 787, 1240, 546]  32.808333  
7                2    [41, 760, 1016, 292]  26.881666  
15               2   [244, 121, 1588, 495]  42.015833  
16               2   [956, 430, 1074, 460]  28.416250  
18               2   [979, 372, 1419, 548]  37.544375  
...            ...                     ...        ...  
99472            2  [1310, 446, 1117, 380]  29.553958  
99493            2   [1442, 735, 967, 471]  25.585208  
99497            2   [261, 417, 1500, 576]  39.687500  
99500            2   [983, 548, 1425, 477]  37.703125  
99506            2  [1014, 576, 1170, 366]  30.956250  

[16057 rows x 6 columns]

In [24]:
dtset4_참돔 = df_data4[(df_data4['category_id']==3)&(df_data4['fish_size']>=24)]
dtset4_참돔

file_name  width  height  \
2      139221_objt_rs_2020-12-17_10-57-19-33_002.JPG   2704    1520   
3      214655_objt_rs_2021-01-04_12-12-53-33_002.jpg   2704    1520   
4      318109_objt_rs_2021-01-18_13-46-48-00_002.JPG   2704    1520   
14     159375_objt_rs_2020-12-10_12-52-00-00_004.JPG   2704    1520   
19     133388_objt_rs_2020-12-16_14-24-32-66_003.JPG   2704    1520   
...                                              ...    ...     ...   
99909  160565_objt_rs_2020-12-10_13-54-26-00_001.JPG   2704    1520   
99915  170096_objt_rs_2020-12-10_14-24-32-33_002.JPG   2704    1520   
99916  169339_objt_rs_2020-12-10_14-55-08-00_001.JPG   2704    1520   
99920   90641_objt_rs_2020-12-05_12-59-16-00_001.JPG   2704    1520   
99946  166789_objt_rs_2020-12-10_12-31-26-33_002.JPG   2704    1520   

       category_id                    bbox  fish_size  
2                3  [1088, 499, 1425, 516]  37.703125  
3                3    [56, 698, 1127, 453]  29.818541  
4                3   [368, 329, 1030, 391]  27.252083  
14               3   [979, 362, 1198, 534]  31.697083  
19               3  [1096, 749, 1063, 419]  28.125208  
...            ...                     ...        ...  
99909            3   [111, 482, 1528, 540]  40.428333  
99915            3  [1321, 806, 1164, 448]  30.797500  
99916            3    [201, 63, 1300, 633]  34.395833  
99920            3    [558, 722, 952, 421]  25.188333  
99946            3   [436, 185, 1507, 565]  39.872708  

[17204 rows x 6 columns]

In [25]:
dtset4_감성돔 = df_data4[(df_data4['category_id']==4)&(df_data4['fish_size']>=25)]
dtset4_감성돔

file_name  width  height  \
9       314049_objt_bp_2021-01-20_10-35-43-66_003.JPG   2704    1520   
22      190108_objt_bp_2020-12-22_12-19-12-66_003.JPG   2704    1520   
25      200044_objt_bp_2020-12-22_16-29-16-33_002.JPG   2704    1520   
31      199702_objt_bp_2020-12-23_17-36-25-33_002.JPG   2704    1520   
35      311958_objt_bp_2021-01-19_15-33-50-66_003.JPG   2704    1520   
...                                               ...    ...     ...   
100282  634677_objt_bp_2021-02-08_09-11-16-66_003.JPG   2704    1520   
100284  634744_objt_bp_2021-02-08_09-13-00-66_006.JPG   2704    1520   
100285  634921_objt_bp_2021-02-08_09-16-37-33_002.JPG   2704    1520   
100287  634638_objt_bp_2021-02-08_09-12-35-33_002.JPG   2704    1520   
100288  634616_objt_bp_2021-02-08_09-10-00-66_006.JPG   2704    1520   

        category_id                    bbox  fish_size  
9                 4   [571, 776, 1268, 606]  33.549166  
22                4  [1177, 562, 1306, 596]  34.554583  
25                4   [890, 576, 1119, 513]  29.606875  
31                4    [94, 692, 1481, 675]  39.184791  
35                4   [223, 187, 1016, 476]  26.881666  
...             ...                     ...        ...  
100282            4   [713, 832, 1097, 446]  29.024791  
100284            4   [905, 289, 1044, 379]  27.622500  
100285            4    [43, 625, 1421, 625]  37.597291  
100287            4  [1039, 411, 1183, 448]  31.300208  
100288            4   [193, 361, 1112, 384]  29.421666  

[18456 rows x 6 columns]

In [26]:
dtset4_돌돔 = df_data4[(df_data4['category_id']==5)&(df_data4['fish_size']>=24)]
dtset4_돌돔

file_name  width  height  \
8       390482_objt_rb_2021-01-21_23-33-39-66_004.JPG   2704    1520   
10      279366_objt_rb_2021-01-16_18-29-33-00_001.JPG   2704    1520   
11      494347_objt_rb_2021-01-30_10-53-13-33_002.JPG   2704    1520   
17      374244_objt_rb_2021-01-06_09-07-57-66_003.JPG   2704    1520   
27      263986_objt_rb_2021-01-16_11-19-51-00_002.JPG   2704    1520   
...                                               ...    ...     ...   
100274  216146_objt_rb_2021-01-05_11-23-57-66_003.jpg   2704    1520   
100275  280577_objt_rb_2021-01-15_10-11-19-00_001.JPG   2704    1520   
100276  228334_objt_rb_2020-12-29_16-58-03-00_001.jpg   2704    1520   
100277  372716_objt_rb_2021-01-23_15-09-43-00_001.JPG   2704    1520   
100278  272988_objt_rb_2021-01-15_11-33-44-33_002.JPG   2704    1520   

        category_id                   bbox  fish_size  
8                 5  [453, 703, 1429, 634]  37.808958  
10                5  [440, 501, 1559, 617]  41.248541  
11                5  [200, 701, 1537, 673]  40.666458  
17                5  [988, 709, 1514, 642]  40.057916  
27                5  [743, 265, 1539, 684]  40.719374  
...             ...                    ...        ...  
100274            5  [594, 271, 1404, 766]  37.147500  
100275            5   [217, 79, 1482, 774]  39.211250  
100276            5  [330, 500, 1355, 687]  35.851041  
100277            5  [819, 739, 1179, 647]  31.194375  
100278            5  [194, 332, 1411, 676]  37.332708  

[13622 rows x 6 columns]

In [27]:
dtset4_광어.to_csv('dtset4_광어.csv' , index=False)
dtset4_우럭.to_csv('dtset4_우럭.csv' , index=False)
dtset4_참돔.to_csv('dtset4_참돔.csv' , index=False)
dtset4_감성돔.to_csv('dtset4_감성돔.csv' , index=False)
dtset4_돌돔.to_csv('dtset4_돌돔.csv' , index=False)

In [127]:
import random
import os
import pandas as pd
import glob
import shutil

In [128]:
total_files = os.listdir('/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4')
len(total_files)

100506

In [135]:
data_root_path_1 = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/dtset4_광어/'
data_root_path_2 = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/dtset4_우럭/'
data_root_path_3 = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/dtset4_참돔/'
data_root_path_4 = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/dtset4_감성돔/'
data_root_path_5 = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/dtset4_돌돔/'

In [137]:
total_files

['120076_objt_rs_2020-12-15_12-33-54-33_002.JPG',
 '442975_objt_of_2021-01-25_11-25-27-00_001.JPG',
 '398618_objt_rb_2021-01-21_21-57-37-66_004.JPG',
 '599170_objt_rb_2021-02-05_09-48-46-66_003.JPG',
 '206165_objt_bp_2020-12-23_17-07-58-33_002.JPG',
 '449606_objt_of_2021-01-26_09-42-07-66_003.JPG',
 '457126_objt_of_2020-12-22_12-28-42-00_001.JPG',
 '446968_objt_of_2021-01-26_09-30-52-66_003.JPG',
 '160445_objt_rs_2020-12-10_11-48-53-00_001.JPG',
 '447540_objt_rb_2021-01-30_13-47-42-00_001.JPG',
 '141209_objt_rs_2020-12-17_11-24-18-00_001.JPG',
 '425516_objt_of_2021-01-20_09-19-08-00_001.JPG',
 '344193_objt_bp_2021-01-20_12-12-47-00_002.JPG',
 '604063_objt_rb_2021-02-05_09-59-39-66_004.JPG',
 '597310_objt_rb_2021-02-05_09-40-11-33_003.JPG',
 '356120_objt_rs_2021-01-17_17-16-52-00_001.JPG',
 '187416_objt_bp_2020-12-22_14-20-56-00_001.JPG',
 '337152_objt_rs_2021-01-18_14-55-34-00_001.JPG',
 '626829_objt_bp_2021-01-20_15-15-37-33_002.JPG',
 '205424_objt_bp_2020-12-23_12-11-50-33_002.JPG',


In [138]:
dtset4_광어

file_name  category_id  fish_size
96     546105_objt_of_2021-02-04_09-08-11-33_002.JPG            1  46.169791
171    571383_objt_of_2021-02-04_10-36-22-00_001.JPG            1  35.004375
288    446970_objt_of_2021-01-26_09-32-39-00_001.JPG            1  36.274375
440    472639_objt_of_2021-01-28_11-20-24-00_001.JPG            1  38.020625
449    458711_objt_of_2020-12-22_13-38-02-33_002.JPG            1  38.629166
...                                              ...          ...        ...
92439  436161_objt_of_2020-12-22_10-12-08-00_001.JPG            1  38.258750
92547  498778_objt_of_2021-02-03_10-16-04-66_003.JPG            1  35.586458
92956  589882_objt_of_2021-02-06_09-01-05-00_001.JPG            1  40.243124
93854  547135_objt_of_2021-02-04_09-11-29-33_002.JPG            1  39.290625
93960  545972_objt_of_2021-02-04_09-02-05-00_001.JPG            1  36.486041

[687 rows x 3 columns]

In [139]:
src = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/'

for i in dtset4_광어['file_name']:
    shutil.copy(src+i , data_root_path_1)

In [140]:
src = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/'

for i in dtset4_우럭['file_name']:
    shutil.copy(src+i , data_root_path_2)

In [141]:
src = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/'

for i in dtset4_참돔['file_name']:
    shutil.copy(src+i , data_root_path_3)

In [142]:
src = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/'

for i in dtset4_감성돔['file_name']:
    shutil.copy(src+i , data_root_path_4)

In [143]:
src = '/Users/kyeongmialicekim/Desktop/AiHub_어종5/어류 개체 촬영 영상/Training/dtset4/'

for i in dtset4_돌돔['file_name']:
    shutil.copy(src+i , data_root_path_5)